helm/daemon/templates/daemon.yaml

```diff
@@ -0,0 +1,256 @@
+            secretKeyRef:
+              name: {{ $node.name }}
+              key: "libp2p-password"
+        {{- end }}
+        {{- if $node.minaKeys.enabled }}
+        - name: MINA_PRIVKEY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: {{ $node.name }}
+              key: "key-password"
+        {{- end }}
+        ports:
+        - name: client-port
+          protocol: TCP
+          containerPort: {{ $node.ports.client }}
+        - name: graphql-port
+          protocol: TCP
+          containerPort: {{ $node.ports.graphql }}
+        - name: metrics-port
+          protocol: TCP
+          containerPort: {{ $node.ports.metrics }}
+        - name: external-port
+          protocol: TCP
+          containerPort: {{ $node.ports.p2p }}
+          hostPort: {{ $node.ports.external }}
+{{$data := dict "name" $node.name "healthcheck" $.Values.healthcheck }}
+{{- include "healthcheck.seed.allChecks" $data | indent 8 }}
+        imagePullPolicy: Always
+        volumeMounts:
+        {{- if $node.libp2pKeys.enabled }}
+        - name: fixed-libp2p-keys
+          mountPath: /root/libp2p-keys
+        {{- end }}
+        {{- if $node.minaKeys.enabled }}
+        - name: fixed-mina-keys
+          mountPath: /root/mina-keys
+        {{- end }}
+        {{- if $.Values.deployment.uploadBlocksToGCloud }}
+        - name: gcloud-keyfile
+          mountPath: "/gcloud/"
+        {{- end }}
+        {{- if $.Values.deployment.runtimeConfig }}
+        - name: daemon-config
+          mountPath: "/config/"
+        {{- end }}
+      volumes:
+      {{- if $.Values.deployment.runtimeConfig }}
+      - name: daemon-config
+        configMap:
+          name: daemon-config
+      {{- end }}
+      {{- if $node.minaKeys.enabled }}
+      - name: mounted-mina-keys
+        secret:
+          secretName: {{ $node.name }}
+          defaultMode: 0600
+          items:
+          - key: {{ $node.name }}-key
+            path: {{ $node.name }}-key
+          - key: {{ $node.name }}-key.pub
+            path: {{ $node.name }}-key.pub
+      - name: fixed-mina-keys
+        emptyDir: {}
+      {{- end }}
+      {{- if $node.libp2pKeys.enabled }}
+      - name: mounted-libp2p-keys
+        secret:
+          secretName: {{ $node.name }}
+          defaultMode: 0600
+          items:
+          - key: {{ $node.name }}-libp2p
+            path: {{ $node.name }}-libp2p
+          - key: {{ $node.name }}-libp2p.peerid
+            path: {{ $node.name }}-libp2p.peerid
+      - name: fixed-libp2p-keys
+        emptyDir: {}
+      {{- end }}
+      {{- if $node.uploadBlocksToGCloud }}
+      - name: gcloud-keyfile
+        secret:
+          secretName: gcloud-keyfile
+          defaultMode: 0400
+          items:
+          - key: keyfile
+            path: keyfile.json
+      {{- end }}
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ $.Values.deployment.name }}-{{ $node.name }}
+spec:
+  {{- if $node.ip }}
+  type: LoadBalancer
```

Comment:
If I read correctly this creates loadBalancer type only if IP is provided. Otherwise provisions as ClusterIP.
Can there be a case when you will want a loadBalancer but with auto assigned ip?
i.e.
```spec:
  {{- if $.node.loadBalancerIP }}
  type: LoadBalancer
  {{- if not (eq $.node.loadBalancerIP "auto") }}
  loadBalancerIP: {{ $.node.loadBalancerIP }}
  {{- end }}
  {{- else }}
  type: ClusterIP
  {{- end }}

Reply:
iirc the way we set it up you basically get this behavior, load balancers require an IP but google actually assigns one automatically so it is just "set loadBalancerIP = anything", get a loadbalancer, else get a ClusterIP. Maybe we should test this and/or be more explicit about it in the naming and use, but that's my recollection on how this works.

Reply:
Maybe we were doing heavy lifting here in terraform, but I have never done anything particularly manual or complicated with it at deploy time in helm.